File: cpp/july/day_11_ Subsets.cpp
class Solution {
public:
    void sub(vector<vector<int>> &res, vector<int> &nums, int pos, vector<int> &t){
        res.push_back(t);
        for(int i = pos; i < nums.size(); i++){
            t.push_back(nums[i]);
            
            sub(res, nums, i+1, t);
            t.pop_back();   
        }
        
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> t;
        sub(res, nums, 0, t);
        return res;
    }
};

File: python3/july/day_11_ Subsets.py
class Solution:
    def sub(self, nums, pos, t):
        self.res.append(t[::])
        for i in range(pos, len(nums)):
            t.append(nums[i])
            self.sub(nums, i+1, t)
            t.pop()

    
    def subsets(self, nums: List[int]) -> List[List[int]]:
        self.res = []
        t = []
        self.sub(nums, 0, t)
        return self.res
    
